internal/controller/controllers/common.go

```diff
@@ -34,22 +34,33 @@ const (
 	mirrorRegistryRefCertKey         = "ca-bundle.crt"
 	mirrorRegistryRefRegistryConfKey = "registries.conf"
 	mirrorRegistryConfigVolume       = "mirror-registry-config"
+	WatchResourceLabel               = "agent-install.openshift.io/watch"
+	WatchResourceValue               = "true"
 )
 
-func getSecret(ctx context.Context, c client.Client, r client.Reader, namespace, name string) (*corev1.Secret, error) {
-	secret := &corev1.Secret{}
-	key := types.NamespacedName{
-		Namespace: namespace,
-		Name:      name,
+func ensureSecretLabel(ctx context.Context, c client.Client, r client.Reader, key types.NamespacedName) error {
```

Comment:
Since labeling is just an implementation detail of how we are getting the secret (with a selective cache), WDYT about moving this entirely within the below `getSecret` function? That way any code in this package can just call `getSecret` and not worry about the cache or label stuff. Within `getSecret`:

* try to get from cache
* if not found, get from API and add the label

Reply:
yes, sounds better. Fixed.